Make running jekyll less irritating

bcl opened the pull request:
Every time I need to edit the blog and view it locally something in the
ruby/jekyll stack has changed enough that it doesn't work. So make a
Dockerfile and some instructions to make setting up the environment more
reliable. And separate from the host.

brlane-rht committed:
Make running jekyll less irritating

Every time I need to edit the blog and view it locally something in the
ruby/jekyll stack has changed enough that it doesn't work. So make a
Dockerfile and some instructions to make setting up the environment more
reliable. And separate from the host.

dashea commented on StyleGuide.md:
```diff
@@ -2,6 +2,20 @@
+    bundle install --binstubs=/tmp/bin/
+    bundle exec /tmp/bin/jekyll serve --host=0.0.0.0 --incremental
+
+...then just open http://localhost:4000/ and you're off. On subsquant runs you can reuse the container with:
```
"subsquant"

dashea commented:
Can the bundle install and bundle exec parts be part of the Dockerfile? Either way it beats what we've got now, so looks good to me besides the spelling error

dashea commented:
Well, there's also this, which matches what happened when I tried to run the bundle stuff locally. I have no clue why it would be different for me:

```
[user@7571b664425f weldr.io]$ bundle exec /tmp/bin/jekyll serve --host=0.0.0.0 --incremental
Ignoring commonmarker-0.17.9 because its extensions are not built. Try: gem pristine commonmarker --version 0.17.9
Ignoring eventmachine-1.2.7 because its extensions are not built. Try: gem pristine eventmachine --version 1.2.7
Ignoring ffi-1.9.25 because its extensions are not built. Try: gem pristine ffi --version 1.9.25
Ignoring http_parser.rb-0.6.0 because its extensions are not built. Try: gem pristine http_parser.rb --version 0.6.0
Ignoring nokogiri-1.8.2 because its extensions are not built. Try: gem pristine nokogiri --version 1.8.2
Could not find commonmarker-0.17.9 in any of the sources
Run `bundle install` to install missing gems.
```

If I run those gem pristine commands (with sudo), it works

bcl commented:
> If I run those gem pristine commands (with sudo), it works

This continues to be nuts. That's exactly the kind of crap this was supposed to avoid.

bcl commented:
> Can the bundle install and bundle exec parts be part of the Dockerfile?

My goal was to make this the reliable environment for running the bundle stuff inside of, since it depends on whatever's in the current git repo I think it makes sense to leave that setup for later. If we include it I expect the new failure mode to be that the docker image gets out of date and we end up debugging that instead. But given how much of a mess this already is, maybe it would be better? I don't know.

brlane-rht committed:
Make the container build instructions more reproducible

Add --distribution so that it will use the .lock file contents,
otherwise with the latest fedora:latest container it fails with a bunch
of complaints.

Add vendor to the _config.yml so that jekyll doesn't think it's part of
your site and try to parse things it shouldn't.

bcl commented:
Reproduced your problem by removing my fedora:latest image and re-building. Added --deployment which makes it use the .lock file. Which isn't ideal, but it works. For the moment.
